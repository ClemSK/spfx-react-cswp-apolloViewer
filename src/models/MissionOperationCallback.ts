import { IMission } from '../IMission';

export type MissionOperationCallback = (mission: IMission) => void;
