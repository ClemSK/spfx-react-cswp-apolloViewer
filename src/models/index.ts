export * from './IMission';
export * from './MissionOperationCallback';
