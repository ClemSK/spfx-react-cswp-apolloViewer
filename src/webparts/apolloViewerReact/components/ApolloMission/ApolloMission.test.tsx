import * as React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';

import { IMission } from '../../../../models';
import { ApolloMission } from './';

let _mission: IMission = undefined;
let _onRemoveMissionCallback: jest.Mock;

beforeAll(() => {
  _mission = {
    id: 'AS-510',
    name: 'Apollo 15',
    launch_date: '07/26/1971',
    end_date: '08/07/1971',
    image: 'apollo-15.png',
    wiki_href: 'https://en.wikipedia.org/wiki/Apollo_15',
    summary:
      'First Extended LM and rover, landed in Hadley-Apennine. Surface EVA time:18:33 hr. Samples returned: 169.10 pounds (76.70 kg).',
    crew: [
      {
        role: 'Commander',
        astronaut_id: '26',
      },
      {
        role: 'Pilot',
        astronaut_id: '17',
      },
      {
        role: 'Sr Pilot',
        astronaut_id: '31',
      },
      {
        role: 'Command Module Pilot',
        astronaut_id: '31',
      },
      {
        role: 'Lunar Module Pilot',
        astronaut_id: '17',
      },
    ],
  };
});

beforeEach(() => {
  // create callback for clicking remove button
  _onRemoveMissionCallback = jest.fn();
});

test('should render mission', () => {
  const wrapper = shallow(
    <ApolloMission
      mission={_mission}
      onRemoveMission={_onRemoveMissionCallback}
    />
  );
  expect(wrapper).toMatchSnapshot();
});

test('raise onRemoveMission event when remove button clicked', () => {
  // create component
  const wrapper = shallow(
    <ApolloMission
      mission={_mission}
      onRemoveMission={_onRemoveMissionCallback}
    />
  );

  // get all anchor tags which are actually butotns
  const buttons = wrapper.find('a');
  expect(buttons.length).toBe(2);

  const removeButton: ShallowWrapper = buttons.at(1);
  removeButton.simulate('click', { preventDefault: () => undefined });

  expect(_onRemoveMissionCallback).toHaveBeenCalled();
  expect(_onRemoveMissionCallback).toHaveBeenCalledTimes(1);
  expect(_onRemoveMissionCallback).toHaveBeenCalledWith(_mission);
});
