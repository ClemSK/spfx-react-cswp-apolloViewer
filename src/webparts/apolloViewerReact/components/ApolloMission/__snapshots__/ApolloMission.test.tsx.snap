// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render mission 1`] = `
<div>
  <table>
    <tbody>
      <tr>
        <td
          className="ms-textAlignRight"
        >
          <strong>
            ID:
          </strong>
        </td>
        <td>
          AS-510
        </td>
      </tr>
      <tr>
        <td
          className="ms-textAlignRight"
        >
          <strong>
            Name:
          </strong>
        </td>
        <td>
          Apollo 15
        </td>
      </tr>
      <tr>
        <td
          className="ms-textAlignRight"
        >
          <strong>
            Date:
          </strong>
        </td>
        <td>
          07/26/1971
           - 
          08/07/1971
        </td>
      </tr>
      <tr>
        <td
          className="ms-textAlignRight"
        >
          <strong>
            Summary:
          </strong>
        </td>
        <td>
          First Extended LM and rover, landed in Hadley-Apennine. Surface EVA time:18:33 hr. Samples returned: 169.10 pounds (76.70 kg).
        </td>
      </tr>
    </tbody>
  </table>
  <button>
    <a
      className="links"
      href="https://en.wikipedia.org/wiki/Apollo_15"
    >
      Learn More
    </a>
  </button>
  <button>
    <a
      className="links"
      href="#"
      onClick={[Function]}
    >
      Remove Mission
    </a>
  </button>
</div>
`;
